includes/class-vindi-payment.php

```diff
@@ -358,18 +358,35 @@ protected function verify_payment_profile($payment_profile_id)
     /**
      * @param array $item
      **/
-  private function return_cycle_from_product_type($item)
+    private function return_cycle_from_product_type($item)
     {
-        if ($item['type'] == 'shipping' || $item['type'] == 'tax')
+        if ($item['type'] == 'shipping' || $item['type'] == 'tax') {
```

Comment:
Acho que vc não precisa dessa verificação `if ($item['type'] == 'shipping' || $item['type'] == 'tax')`, já que vc verifica se é "One Time Shipping" de qualquer forma.

```suggestion
private function return_cycle_from_product_type($item)
{
    if (!$this->is_subscription_type($item->get_product())
        || $this->is_one_time_shipping($item->get_product())) {
        return 1;
    }
    return null;
}
```
Isso não resolveria?